app/src/main/java/dev/leonlatsch/photok/model/io/ThumbnailManager.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ *   Copyright 2020-2024 Leon Latsch
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package dev.leonlatsch.photok.model.io
+
+import android.content.Context
+import android.graphics.Bitmap
+import androidx.core.graphics.drawable.toBitmap
+import coil.ImageLoader
+import coil.decode.VideoFrameDecoder
+import coil.request.ImageRequest
+import coil.size.Scale
+import dev.leonlatsch.photok.model.database.entity.Photo
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+
+
+/** Maximum size of the thumbnail in pixels */
+private const val THUMBNAIL_SIZE = 128
+
+/**
+ * Manages the creation and loading of thumbnails for photos and videos.
+ *
+ * @param context the application context
+ * @param encryptedStorageManager the [EncryptedStorageManager] to access the encrypted files
+ *
+ * @since 1.7.2
+ * @author Starry Shivam
+ */
+class ThumbnailManager(
+    private val context: Context,
+    private val encryptedStorageManager: EncryptedStorageManager
+) {
+    /**
+     * Loads the full size thumbnail stored for this photo as a [ByteArray]
+     *
+     * @param photo the photo to load the thumbnail for
+     * @return the thumbnail as a [ByteArray]
+     */
+    fun loadThumbnail(photo: Photo): ByteArray? {
```

Comment:
This and loadVideoPreview are not used anywhere right? I think we don't need them for now.

Just checked. The old methods are not used anywhere either. So lets get rid of them.

Meaning this could also be a use case with just 1 `invoke()` function named `CreateThumbnailsUseCase()`

```kotlin
class CreateThuzmbnailsUseCase() @Inject constructor(...) {
    suspend operator fun invoke() = suspendCoroutine { continuation 
        // Code here
    }
}
```